shub/image/list.py

```diff
@@ -52,27 +54,38 @@ def list_cmd_full(target, silent, version):
     version = version or config.get_version()
     image_name = utils.format_image_name(image, version)
     target_conf = config.get_target_conf(target)
-    for spider in list_cmd(image_name,
-                           target_conf.project_id,
-                           target_conf.endpoint,
-                           target_conf.apikey):
+    metadata = list_cmd(image_name,
+                        target_conf.project_id,
+                        target_conf.endpoint,
+                        target_conf.apikey)
+    for spider in metadata.get('spiders', []):
         click.echo(spider)
 
 
 def list_cmd(image_name, project, endpoint, apikey):
     """Short version of list cmd to use with deploy cmd."""
     settings = _get_project_settings(project, endpoint, apikey)
-
-    # Run a local docker container to run list-spiders cmd
-    status_code, logs = _run_list_cmd(project, image_name, settings)
-    if status_code != 0:
+    environment = {'JOB_SETTINGS': json.dumps(settings)}
+    exit_code, logs = _run_cmd_in_docker_container(
+            image_name, 'shub-image-info', environment)
+    if exit_code == 0:
+        return _extract_metadata_from_image_info_output(logs)
+    # shub-image-info command not found, fallback to list-spiders
+    elif exit_code == 127:
+        # FIXME we should pass some value for SCRAPY_PROJECT_ID anyway
+        # to handle `scrapy list` cmd properly via sh_scrapy entrypoint
```

Comment:
Earlier `scrapinghub-entrypoint-scrapy` relied on `SCRAPY_PROJECT_ID` to decide between a spider and a python script, later we changed it with https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/pull/26 in version `0.8.8`, but there may be a couple of custom images in-use relying on previous sh-ep-scrapy versions. So if there's no `SCAPY_PROJECT_ID` for such cases, `scrapy list` will be launched incorrectly (as a package script) and fail with not-found-error or something like this.

Now I wonder how many projects use custom images with oldy `scrapinghub-entrypoint-scrapy` versions in real, maybe it's already fine to finally drop it. That's a shame that PyPi disabled download stats :confused: